internal/testrunner/runners/system/runner.go

```diff
@@ -1603,20 +1786,93 @@ func deleteDataStreamDocs(ctx context.Context, api *elasticsearch.API, dataStrea
 	return nil
 }
 
-func filterAgents(allAgents []kibana.Agent, svcInfo servicedeployer.ServiceInfo) []kibana.Agent {
+func filterAgents(allAgents []kibana.Agent, agentInfo agentdeployer.AgentInfo, threshold time.Time, svcInfo servicedeployer.ServiceInfo, runIndependentElasticAgent bool) []kibana.Agent {
+	if agentInfo.Agent.Host.NamePrefix != "" {
+		logger.Debugf("filter agents using agent criteria: NamePrefix=%s", agentInfo.Agent.Host.NamePrefix)
+	}
+
 	if svcInfo.Agent.Host.NamePrefix != "" {
-		logger.Debugf("filter agents using criteria: NamePrefix=%s", svcInfo.Agent.Host.NamePrefix)
+		logger.Debugf("filter agents using service criteria: NamePrefix=%s", svcInfo.Agent.Host.NamePrefix)
+	}
+
+	expectedHostname := agentInfo.Hostname
+	expectedTags := agentInfo.Tags
+	if svcInfo.Agent.Host.NamePrefix == "docker-custom-agent" {
+		// custom agents are still started from servicedeployer, they are defined in the same docker compose scenario
+		expectedHostname = svcInfo.AgentHostname
+		expectedTags = svcInfo.Tags
 	}
 
+	// filtered list of agents must contain all agents started by the stack
+	// they could be assigned the default policy (elastic-agent-managed-ep) or the test policy (ep-test-system-*)
 	var filtered []kibana.Agent
 	for _, agent := range allAgents {
 		if agent.PolicyRevision == 0 {
 			continue // For some reason Kibana doesn't always return a valid policy revision (eventually it will be present and valid)
 		}
 
-		if svcInfo.Agent.Host.NamePrefix != "" && !strings.HasPrefix(agent.LocalMetadata.Host.Name, svcInfo.Agent.Host.NamePrefix) {
+		// It cannot filtered by "elastic-agent-managed-ep" , since this is the default
+		// policy assigned to the agents when they first enroll
+		if agent.PolicyID == "fleet-server-policy" {
+			logger.Debugf("filtered agent (policy id) %q", agent.ID) // TODO: remove
 			continue
 		}
+
+		if agent.Status != "online" {
+			logger.Debugf("filtered agent (not online) %q", agent.ID) // TODO: remove
+			continue
+		}
+
+		if runIndependentElasticAgent && agent.EnrolledAt.Before(threshold) {
```

Comment:
There are many ifs based on `runIndependentElasticAgent`. Should we have two different `filterAgent` methods?

Reply:
Sure, restored the previous `filterAgents` method and created a new one for the new independent agents.